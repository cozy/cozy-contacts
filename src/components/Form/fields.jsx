import React from 'react'
import Input from 'cozy-ui/transpiled/react/Input'
import Textarea from 'cozy-ui/transpiled/react/Textarea'

function adaptField(WrappedComponent) {
  class AdaptField extends React.Component {
    render() {
      const {
        input,
        meta: { valid },
        ...rest
      } = this.props
      return <WrappedComponent {...input} {...rest} data-valid={valid} />
    }
  }
  AdaptField.displayName = `adaptField(${getDisplayName(WrappedComponent)})`
  return AdaptField
}

function getDisplayName(WrappedComponent) {
  return WrappedComponent.displayName || WrappedComponent.name || 'Component'
}

const AdaptedInput = adaptField(Input)
const AdaptedTextarea = adaptField(Textarea)

export { AdaptedInput, AdaptedTextarea }
