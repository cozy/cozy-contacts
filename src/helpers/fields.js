export const addField = fields => fields.push(undefined)

export const removeField = (fields, index) => {
  const isLastRemainignField = fields.length === 1
  fields.remove(index)
  if (isLastRemainignField) addField(fields)
}
