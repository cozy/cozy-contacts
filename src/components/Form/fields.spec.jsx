import React from 'react'
import { Field, Form } from 'react-final-form'
import renderer from 'react-test-renderer'

import { AdaptedInput, AdaptedTextarea } from './fields'

describe('AdaptedInput', () => {
  it('should adapt cozy-ui input for react-final-form', () => {
    const jsx = (
      <Form
        initialValues={{ title: 'Hello' }}
        onSubmit={jest.fn()}
        render={({ handleSubmit }) => (
          <form onSubmit={handleSubmit}>
            <Field name="title" component={AdaptedInput} />
          </form>
        )}
      />
    )
    const tree = renderer.create(jsx).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('AdaptedTextarea', () => {
  it('should adapt cozy-ui input for react-final-form', () => {
    const jsx = (
      <Form
        initialValues={{
          description: 'Nostrum expedita qui rerum non est asperiores ratione.'
        }}
        onSubmit={jest.fn()}
        render={({ handleSubmit }) => (
          <form onSubmit={handleSubmit}>
            <Field name="description" component={AdaptedTextarea} />
          </form>
        )}
      />
    )
    const tree = renderer.create(jsx).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
