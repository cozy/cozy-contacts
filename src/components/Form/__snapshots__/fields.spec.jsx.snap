// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdaptedInput should adapt cozy-ui input for react-final-form 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="styles__c-input-text___3TAv1 styles__c-input-text--large___28EaR"
    data-valid={true}
    name="title"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value="Hello"
  />
</form>
`;

exports[`AdaptedTextarea should adapt cozy-ui input for react-final-form 1`] = `
<form
  onSubmit={[Function]}
>
  <textarea
    className="styles__c-textarea___D7EEH"
    data-valid={true}
    name="description"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder=""
    value="Nostrum expedita qui rerum non est asperiores ratione."
  />
</form>
`;
